responsibleai/responsibleai/databalanceanalysis/data_balance_helper.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+import warnings
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+
+from responsibleai.databalanceanalysis.constants import Constants
+
+FEATURE_BALANCE_MEASURES_KEY = "FeatureBalanceMeasures"
+DISTRIBUTION_BALANCE_MEASURES_KEY = "DistributionBalanceMeasures"
+AGGREGATE_BALANCE_MEASURES_KEY = "AggregateBalanceMeasures"
+
+
+class DataBalanceHelper:
+    """
+    Helper class for computing and transforming data balance measures
+    to be used in a dashboard.
+    """
+
+    @staticmethod
+    def prepare_df(
+        df: pd.DataFrame, target_column: str, pos_label: Optional[str]
+    ) -> pd.DataFrame:
+        """
+        Prepare dataframe for data balance analysis.
+
+        :param df: The dataframe to prepare.
+        :type df: pd.DataFrame
+        :param target_column: The target column in the dataframe.
+        :type target_column: str
+        :param pos_label: The positive label of the target column.
+        :type pos_label: Optional[str]
+        :return: The prepared dataframe.
+        :rtype: pd.DataFrame
+        """
+        try:
+            # Deepcopy the df otherwise the original df will mutate
+            df = df.copy(deep=True)
+
+            # If the positive label is specified, transform the target column
+            # to {0, 1} because Data Balance Analysis only supports binary
+            # classification for now.
```

Comment:
Are we surfacing that limitation somewhere? For example, is there an error warning the user if we have, say, a regression task?

Reply:
So right now, the `DataBalanceManager._validate` method raises an error if the target column contains != 2 unique values.
I thought it was good to do it this way rather than checking `task_type`. What do you think?

The snippet is:
```
if self._df[self._target_column].unique().size != 2:
            raise ValueError(
                (
                    f"The target_column '{self._target_column}' must contain"
                    " only a positive label and a negative label to compute"
                    " data balance measures."
                )
            )
```

Reply:
I think `task_type` is safer, although I would have agreed with you a couple of years ago 🙂  Since then I have run into too many situations where the test or training set somehow randomly ended up with only 1s or only 0s, so suddenly you have only 1 label and things fail.

Reply:
Is it okay if I do both checks -- `task_type == classification` and `target_column.unique().size == 2`?
This way, it would specifically check for binary classification 
However, the user can have a dataset with all 0s or 1s in the label column even if it is a binary classification scenario... hmm, thoughts?

Reply:
You can check for task_type and `size <= 2` 🙂

Reply:
Done! 😊